mwdb/core/search/fields.py

```diff
@@ -253,20 +253,17 @@ def get_condition(self, expression: Expression, remainder: List[str]) -> Any:
         value = get_term_value(expression)
 
         def jsonpath_escape(field):
-            # Escapes field to be correctly represented in jsonpath
-            asterisk = ""
+            """Escapes field to be correctly represented in jsonpath"""
             # Escape all double quotes
             field = field.replace('"', '\\"')
-            if field.endswith("*"):
-                # If asterisk is escaped: remove the escape
-                if field[-2] == "\\" and field[-3] != "\\":
-                    # aaaa\* -> "aaaa*"
-                    field = field[:-2] + "*"
-                else:
-                    # aaaa* -> "aaaa"[*]
-                    asterisk = "[*]"
-                    field = field[:-1]
-            return f'"{field}"{asterisk}'
+            # Find trailing non-escaped asterisks
+            asterisk_r = re.search(r"(?<!\\)([*]+)$", field)
+            asterisk_levels = len(asterisk_r.group(0)) if asterisk_r else 0
+            field = field[:-asterisk_levels]
+            asterisks = asterisk_levels * "[*]"
+            # Unescape all escaped asterisks
+            field = re.sub(r"\\[*]", "*", field)
+            return f'"{field}"{asterisks}'
```

Comment:
Does the field necessarily have to be quoted here?
I don't think this is a problem and it helps in some cases, but I'd rather point out to it.

Reply:
Yes, it's necessary if field contains quotes that must be escaped.